StarTrek13/Xel/borg_structures.dm

```diff
@@ -199,55 +201,168 @@
-	desc = "Symbols flash on its holographic display as it constantly flickers and hums, used to fly ships it would seem."
+	desc = "Symbols flash on its holographic display as it constantly flickers and hums."
+	icon = 'StarTrek13/icons/borg/borg.dmi'
 	icon_state = "navicomp"
-	anchored = 1
-//	pixel_x = -32
+	pixel_x = -32
 	bound_width = 96
 	layer = 4.5
-	parts = list(
-							/obj/item/stock_parts/borg/bin = 2,
-							/obj/item/stock_parts/borg/capacitor = 1,
-							/obj/item/stock_parts/borg = 3)
 
-/obj/machinery/borg/helm/New()
-	. = ..()
-	var/area/A = get_area(src)
-	if(SSticker.mode.hivemind.borg_machines_room_has_nav == 0)
-		if(istype(A, SSticker.mode.hivemind.borg_target_area))
-			SSticker.mode.hivemind.borg_machines_room_has_nav = 1
-		//	src.say(SSticker.mode.borg_machines_in_area)
-		else
-			src.say("not in the right area")
+/obj/item/borg
+	name = "DEBUG"
+	desc = "Huh.. You've never seen anything like this before."
+	icon = 'StarTrek13/icons/borg/borg_items.dmi'
+	var/cons_time
+
+/obj/item/borg/manip
+	name = "manipulator"
+	icon_state = "manipulator"
+	cons_time = 2
+
+/obj/item/borg/amp
+	name = "amplifier"
+	icon_state = "amplifier"
+	cons_time = 2
+
+/obj/machinery/borg/factory //BROKEN.
+	name = "borg assembly line"
+	desc = "A borg assembly line. It fabricates parts at immense speeds."
+	icon_state = "factory"
+	bound_width = 96
+	layer = 5.5
+	var/running
+	var/part_time = 20 //DEBUG. currently set for two seconds PER.//per part.
+	var/required_points = 0 // How many assimilation points to construct this item?
+	var/constructables = list(/obj/item/borg/manip, /obj/item/borg/amp) //What can this factory build"?
+	//var/construct_amount = 4 //How many per cycle?
+	var/constructing = null
+	var/saved_time
+
+/obj/machinery/borg/factory/Destroy()
+	//STOP_PROCESSING(SSobj, src)
+	construction(TRUE)
+	..()
+
+/obj/machinery/borg/factory/update_icon()
+	if(running)
+		icon_state = "factory_on"
 	else
-		src.say("there is already one of those here")
-		qdel(src)
+		icon_state = "factory"
+
+/obj/machinery/borg/factory/attack_hand(mob/living/carbon/human/user)
+	if(!isborg(user))
+		return
+	var/query
+	query = input(user, "Select an item for production.","Select Production",query) as anything in constructables|"cancel production" // Yes, I know this lists the item paths, not names. It's temporary.
+	if(!query)
+		return
+	if(SSticker.mode.hivemind.cons_points < required_points)
+		to_chat(user, "<span class='warning'>We do not have enough construction materials to fabricate this part.</span>")
+		return
+	if(query == "cancel production")
+		construction(TRUE)
+		update_icon()
+		return
+	to_chat(user, "<span class='info'>The [src] is now producing [query].</span>")
+	constructing = query
+	//part_time = query.cons_time * 10
+	saved_time = world.time + part_time
+	construction()
+	update_icon()
+
+/obj/machinery/borg/factory/proc/construction(var/stop)
+	to_chat(world, "PROCESSING A BORG FACTORY")
```

Comment:
HI debug code

Reply:
Factory is more-or-less broken, it shouldn't really be spawned anyways?

Reply:
then why is it in this PR?

Reply:
Good question